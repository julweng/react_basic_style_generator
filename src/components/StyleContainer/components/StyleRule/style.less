@import "~styles/common.less";

.style__rule__container {
  display: flex;
  flex-direction: column;

  div {
    display: block;
    margin: 0 auto;
    p {  
      margin: 0.3rem;
      text-align: left;
      font-size: 1.5rem;
      padding: 0;
      font-family: @defaultFont;
    }
  }
}
