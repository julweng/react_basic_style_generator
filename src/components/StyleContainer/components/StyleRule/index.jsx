import React from "react"
import { string, shape, number } from "prop-types"
import { generateCssRule } from "../../functions"
import "./style.less"

const StyleRule = ({
  style: {
    color = "",
    backgroundColor = "",
    fontFamily = "",
    fontWeight = "",
    fontStyle = ""
  }
}) => {
  const font = generateCssRule(fontFamily, "font family")
  const textColor = generateCssRule(color, "font color")
  const background = generateCssRule(backgroundColor, "background color")
  const style = generateCssRule(fontStyle, "font style")
  const weight = generateCssRule(fontWeight, "font weight")

  return (
    <div className="style__rule__container">
      <div>
        <p>{`color: ${textColor};`}</p>
        <p>{`background-color: ${background};`}</p>
        <p>{`font-family: ${font};`}</p>
        <p>{`font-style: ${style};`}</p>
        <p>{`font-weight: ${weight};`}</p>
      </div>
    </div>
  )
}

StyleRule.propTypes = {
  style: shape({
    color: string,
    backgroundColor: string,
    fontFamily: string,
    fontWeight: number,
    fontStyle: string
  })
}
export default StyleRule
