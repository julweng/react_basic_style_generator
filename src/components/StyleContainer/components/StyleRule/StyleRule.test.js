import React from "react"
import { render, cleanup } from "@testing-library/react"
import "jest-dom/extend-expect"
import StyleRule from "./index"

const createProps = ({
  style = {
    color: "#000",
    backgroundColor: "#FFF",
    fontFamily: "Arial, sans-serif",
    fontWeight: 400,
    fontStyle: "normal"
  }
} = {}) => ({
  style
})

describe("StyleRule", () => {
  afterEach(cleanup)

  it("Should render without crashing", () => {
    const props = createProps()
    const { container } = render(<StyleRule {...props} />)
    expect(container).toBeTruthy()
    expect(container).toMatchSnapshot()
  })

  it("Should display 7 p tags for the css rule", () => {
    const props = createProps()
    const { container } = render(<StyleRule {...props} />)
    const p = container.querySelectorAll("p")
    expect(p.length).toEqual(5)
  })

  it("Should display the css rule", () => {
    const props = createProps()
    const { container } = render(<StyleRule {...props} />)
    const p = container.querySelectorAll("p")
    expect(p[0]).toHaveTextContent(`color: "#000";`)
    expect(p[1]).toHaveTextContent(`background-color: "#FFF";`)
    expect(p[2]).toHaveTextContent(`font-family: "Arial", sans-serif;`)
    expect(p[3]).toHaveTextContent(`font-style: "normal";`)
    expect(p[4]).toHaveTextContent(`font-weight: 400;`)
  })

  it("Should display the default message if no style rules are available", () => {
    const props = createProps({ style: {} })
    const { container } = render(<StyleRule {...props} />)
    const p = container.querySelectorAll("p")
    expect(p[0]).toHaveTextContent("select a font color")
    expect(p[1]).toHaveTextContent("select a background color")
    expect(p[2]).toHaveTextContent("select a font family")
    expect(p[3]).toHaveTextContent("select a font style")
    expect(p[4]).toHaveTextContent("select a font weight")
  })
})
