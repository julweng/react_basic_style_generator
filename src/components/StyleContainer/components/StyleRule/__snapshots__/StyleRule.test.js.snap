// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleRule Should render without crashing 1`] = `
<div>
  <div
    class="style__rule__container"
  >
    <div>
      <p>
        color: "#000";
      </p>
      <p>
        background-color: "#FFF";
      </p>
      <p>
        font-family: "Arial", sans-serif;
      </p>
      <p>
        font-style: "normal";
      </p>
      <p>
        font-weight: 400;
      </p>
    </div>
  </div>
</div>
`;
